### ISSUE TITLE
issue: custom field component for a type z.number().int() is ignored when I have z.number() present

### ISSUE THREAD

**Issue by capaj** [iway1/react-ts-form#117]
### Version Number

1.6.4

### Steps to reproduce

I have this mapping
 ```ts
export const JsonSchema = z.object({})
export const NullableDate = z.date().nullable()
export const CategoryId = z
  .number()
  .int()
  .refine((val) => val > 0)

const mapping = [
  [z.string(), TextField],
  [z.boolean(), CheckboxField],
  [z.number(), NumberField],
  [z.nativeEnum(AdminUserRoles), makeSelectFieldForEnum(AdminUserRoles)],
  [JsonSchema, JsonSchemaField] as const,
  [NullableDate, DateField] as const,
  [CategoryId, CategoryIdField] as const
] as const // 👈 `as const` is necessary

```

yet even when I use `CategoryId` in my zod schema, it is still rendering with `NumberField`

I will try to prepare a code sandbox

### Expected behaviour

It should prefer the exact zod type rather than the generic `z.number()` field component

**Comment by iway1** [comment]
definite bug, checking this out

**Comment by iway1** [comment]
I think this is what we want the behavior to be, since the schema is just a `z.number()` with some extra conditions - when checking against the mapping we just look for the zod type of the schema. This allows adding additional checks at the schema level

Should be able to use `createUniqueFieldSchema()` on `CategoryId` to make it map correctly:

```ts
export const CategoryIdSchema = createUniqueFieldSchema('category-id',  z
 .number()
 .int()
 .refine((val) => val > 0)
 )
```

This seems consistent with how other types work in the library - let me know if you think there's a reason `z.number().int()` should be considered as a something different from `z.number().int()` though. I'll leave open for now for discussion

**Comment by scamden** [comment]
I agree. Can always use `createUniqueFieldSchema` to force a separation as well right?
